implot_internal.h

```diff
@@ -1173,30 +1192,17 @@ struct ImPlotNextPlotData
 
 // Temporary data storage for upcoming item
 struct ImPlotNextItemData {
-    ImVec4          Colors[5]; // ImPlotCol_Line, ImPlotCol_Fill, ImPlotCol_MarkerOutline, ImPlotCol_MarkerFill, ImPlotCol_ErrorBar
-    float           LineWeight;
-    ImPlotMarker    Marker;
-    float           MarkerSize;
-    float           MarkerWeight;
-    float           FillAlpha;
-    float           ErrorBarSize;
-    float           ErrorBarWeight;
-    float           DigitalBitHeight;
-    float           DigitalBitGap;
+    ImPlotSpec      Spec;
```

Comment:
I'm not that familiar with the internals, but what is ImPlotNextItemData's lifetime relative to the user-provided ImPlotSpec?  I ask because storing this by value forces a memcpy of the passed-in spec.  This is probably the safest thing to do, but does add a bit of overhead--storing a pointer would avoid the copy, but if lifetimes could differ (or user code could modify their ImPlotSpec in between uses of NextItemData), using a pointer would definitely be unsafe.

Reply:
`ImPlotNextItemData` is a relic of the `SetNext` API (those functions would set variables on the current `ImPlotNextItemData `). I was tempted to remove it, but it's still doing a few useful things. Particularly, it stores transient flags about what should be rendered (e.g. `bool RenderLine` etc. which are established in `BeginItem`) and a transient `ImPlotSpec Spec`. This spec may or may not be exactly what the user passed! For instance, if the user-passed spec has any automatic variables, then `BeginItem` will convert them to whatever is appropriate and set `ImPlotNextItemData::Spec` accordingly. Thus, all the actual rendering code uses `ImPlotNextItemData::Spec` instead of the user provided spec. It's a little weird, admittedly, but it's functional.

The alternative, that I see, is that we pass specs to `PlotX` by non-const reference or pointer, update automatic variables in place, and use that throughout the render pipleine. Either of those would make the API less convenient than it currently is, and I think changing the user provided spec would cause confusion/issues.

Reply:
Thanks, the current approach in this PR makes sense in that context.